### DISCUSSION TITLE
High resolution screens

### DISCUSSION THREAD

**Discussion by stefonarch** [lxqt/lxqt#2054] [General]
### Background
I had to change my laptop, similar model, but ATI driver and higher resolution. I put together here what I tried, what happened and most what isn't perfect.

```
$ xdpyinfo |grep -B2 resolution
screen #0:
  dimensions:    1920x1080 pixels (508x285 millimeters)
  resolution:    96x96 dots per inch

```

As I inserted my previous disk which contained a "perfect" system all fonts were really really tiny, so I had to change them from 10 to 13/14 everywhere:
* desktop preferences
* lxqt-appearence
* qterminal
* gps
* featherpad/notes
* plugin-customcommand
* firefox & thunderbird
 
Same goes for icon sizes, thumbnail sizes ecc in pcmanfm-qt and panel/icon dimensions.

#### Results
everything fine, except reading some mails in txt in thunderbird and some webpages which are still too tiny, so always cltr++ which is annoying.

### The other way round, changing dpi values

* on the fly to test, restart applications after:
`xrandr --dpi 120' 
(steps should be 96, 120 (25%), 144 (50%), becaus of pixmaps icons if I'm right)

* in `/etc/X11/xorg.conf.d/90-monitor.conf `
```
 Section "Monitor"
    Identifier             "eDP"
    DisplaySize            406 228
    Option                 "DPI" "120 x 120"
EndSection

```
#### Results:
everything fine again with font sizes of 10px, except all in thunderbird/firefox which can be partially solved in `about:config` set to 1.25 instead of -1.0 the value in layout.css.devPixelsPerPx, which solved content font size but not menubar, toolbars and left panel.

#### Solution (probably)
 no dpi configuration, but in mozilla apps use the setting mentioned above whicj will affect content while LXQt will think about the rest.

**Comment by tsujan** [comment]
Is 1920x1080 a high resolution? Maybe it's the highest *before* HDPI. Both my laptops and my external monitor have the same resolution and I didn't have any problem with them.

With real HDPI, you just need to set the environment variable `QT_SCALE_FACTOR` to `2` or `1.5` and do a similar thing with its GTK counterpart (don't remember its name — it's only for gtk). You don't need to touch anything else.

EDIT: The GTK-related EV may be `GDK_SCALE`.

**Reply by stefonarch** [reply]
No it isn't probably real hdpi. I tried scale factor once, forgot to mention it, didn't see any effect in my case. 
This is meant more as a collection of the things which I found, maybe at the end better make a clearer wiki page about that.

On a sidenote, I happen now to have generic icon always for telegram, never happened. 
I compiled the latest lxqt-notificationd do not disturb branch but this shouldn't  affect anything there.

**Reply by tsujan** [reply]
> I tried scale factor once, forgot to mention it, didn't see any effect in my case.

Scale factor is too much for 1920x1080 but it *should* have effect (after log out and log in — don't use it). To try it temporarily, close all FeatherPad windows and enter this in a terminal:

```
QT_SCALE_FACTOR=2 fpad
```
BTW, I have Intel (with modesetting), not ATI.

My font size is 9 but anyone that sees my screen says, "How do you read on this screen?" (I'm nearsighted and have had eyeglasses since childhood). I think, for most people, 10 or 11 is good.

> I compiled the latest lxqt-notificationd do not disturb branch but this shouldn't affect anything there.

No, it doesn't affect Status Notifier. Let's hope something isn't up-to-date on your system ;) Otherwise, after all those changes, Status Notifier may still have a problem (which is possible; see the second paragraph of https://github.com/lxqt/lxqt-notificationd/pull/271#issuecomment-893550824).

**Reply by edwardp3** [reply]
This caught my interest, because I once had LXDE and LXQt installed on an older laptop, it had a high-definition screen, but I also had to enlarge the font size because they were too small on the laptop screen (15"). I did not fault LXDE or LXQt, it was simply a smaller screen. Early on, it used the fglrx driver then went to radeon when fglrx development was discontinued.

From what I have been able to find (via Wikipedia, citing Google Android developer documentation), HDPI is 213-240 dpi.
https://en.wikipedia.org/wiki/Pixel_density#Named_pixel_densities

**Reply by heckert-gamedev** [reply]
I would expect a scaling setting to be configurable either with the monitor settings or in the appearance configuration.
Also, you can set DPI in the appearance settings but I think this should rather be something that belongs with the screen settings...

When I set the font size to be 12pt, button labels didn't fit anymore and the buttons weren't expanded to accomodate the text...

**Reply by tsujan** [reply]
Yes, an option should be added to the appropriate config dialog.

For now, set EVs `QT_SCALE_FACTOR` (for Qt apps) and `GDK_SCALE` (for GTK apps) in LXQt Session Settings if you need scaling.

**Reply by falko** [reply]
I just raised this as issue #2173 after also spending quite some time to figure it out.

**Reply by falko** [reply]
Digging further I found lxqt/lxqt-config#402

**Reply by falko** [reply]
Okay, the [docs briefly mention HIDPI](https://manual.lubuntu.me/stable/3/3.2/3.2.13/session_settings.html?highlight=hidpi) but then link to a [forum post that currently seems to be offline](https://forum.lxqt.org/t/detailed-guide-to-enable-high-dpi-scaling-on-lxqt/507/5). I'll try the hint on `XCURSOR_SIZE` though...

**Reply by falko** [reply]
Reading https://wiki.archlinux.org/title/HiDPI#X_Resources make me wonder if `QT_SCALE_FACTOR` is even the right setting. That wiki page recommends `QT_AUTO_SCREEN_SCALE_FACTOR=1` citing the [Qt documentation on High DPI Displays](https://doc.qt.io/qt-5/highdpi.html). However, that doesn't scale fonts. So it would have to be combined with setting `Xft.dpi`, which brings us back to lxqt/lxqt-config#403

**Reply by falko** [reply]
Looks like `QT_AUTO_SCREEN_SCALE_FACTOR=1` has been [discussed before](https://github.com/lxqt/lxqt-config/issues/401#issuecomment-965061780) and is not a good idea. Sorry, for warming that question up.

**Reply by tsujan** [reply]
> Looks like QT_AUTO_SCREEN_SCALE_FACTOR=1 has been discussed before and is not a good idea.

`QT_AUTO_SCREEN_SCALE_FACTOR` has been removed in Qt6, while `QT_SCALE_FACTOR` works as in Qt5.

EDIT: The new Qt doc is https://doc.qt.io/qt-6/highdpi.html. We still don't know whether LXQt 1.1.0 will be based on Qt6 but setting `QT_SCALE_FACTOR` seems to be the cleanest way of handling HiDPI. With it, no other change is needed.

**Reply by tsujan** [reply]
A screenshot with Qt6 and a `QT_SCALE_FACTOR` of 4 shows how well it works (no change in font or anything else — you may need to open the image in a separate page to see its real size):

![scaling](https://user-images.githubusercontent.com/981076/148461308-fbe81e0b-d609-40b1-9ec1-8a5f0464d0be.png)

**Comment by dieselnutjob** [comment]
Hi I have a 10" tablet with a 1920x1280 pixel screen touchscreen.
I have found that setting QT_SCALE_FACTOR=2 works very nicely, except the title bar (sorry if not the right term) is still tiny.
This makes it very difficult to tap the X to terminate a program.
It seems like a bit of an inconsistency to me.

**Reply by tsujan** [reply]
The window manager is responsible for the title-bar, and it doesn't follow the scale factor. Depending on the WM you use, you could make its font or button bigger.
